File: algorithms/implementation/electronics-shop/electronics-shop-javascript/js/test_getMoneySpent.js
import {getMoneySpent} from '../js/getMoneySpent.js';
import assert from 'assert';

describe('getMoneySpent', function() {

    it('O máximo que pode ser gasto é -1', function() {

        let teclados = ['4']
        let dispositivos = ['5']
        let b = 5

        let esperado = -1;

        let resultado = getMoneySpent(teclados, dispositivos, b);

        assert.strictEqual(resultado, esperado);
    });
});

describe('getMoneySpent', function() {

    it('O máximo que pode ser gasto é 9', function() {

        let teclados = ['3, 1'];
        let dispositivos = ['5, 2, 8'];
        let b = 10;

        let esperado = 9;

        let resultado = getMoneySpent(teclados, dispositivos, b);

        assert.strictEqual(resultado, esperado);
    });
});

File: algorithms/implementation/electronics-shop/electronics-shop-python/tests/test_getMoneySpent.py
from getMoneySpent import getMoneySpent

class TesteGetMoneySpent:
    
    def testeCaso1(self):
        
        teclados  = [3, 1]
        dispositivos = [5, 2, 8]
        b = 10
        
        esperado = 9
        
        resultado = getMoneySpent(teclados, dispositivos, b)
        
        assert resultado == esperado
        
        
    def testeCaso2(self):
        
        teclados  = [4]
        dispositivos = [5]
        b = 5
        
        esperado = -1
        
        resultado = getMoneySpent(teclados, dispositivos, b)
        
        assert resultado == esperado